main.go

```diff
@@ -99,67 +135,67 @@ func main() {
 	deltaList := GetDeltaList(dmwKnownContacts, dfoActiveContacts)
-	fmt.Println(deltaList)
+	var deltaContacts DmwKnownContacts
+	if deltaList != nil {
+		err := json.Unmarshal(deltaList, &deltaContacts)
+		if err != nil {
+			fmt.Println("Cannot unmarshal deltaList")
+		}
+	}
+	fmt.Println(deltaContacts.Contacts)
 
-	//TODO: Update each contact in the deltaList to set state to closed
 	//TODO: Call api to update DMW contact states to set contact to closed.
-
-
+	//for _, contact := range deltaContacts.Contacts {
+	//	var eventType string
+	//	var newState string
+	//	//var trigger fsm.Trigger
+	//	//var err error
+	//	if contact.newState > 0 {
+	//		eventType = CONTACTSTATEUPDATED
+	//	} else {
+	//		eventType = CONTACTENDED
+	//	}
+	//	contact = digiEventUpdateContact(contact.EventID, contact.newState, event.GetCreatedAt(), nil, nil, eventType, action, existingContact)
+	//
+	//}
 }
 
-func GetDeltaList(dmwKnownContacts DmwKnownContacts, dfoActiveContacts DfoActiveContacts) []byte {
-	found := false
-	type DmwKnownContacts []DmwKnownContact
-	deltaList := DmwKnownContacts{}
-
-	// Loop through DmwKnownContacts and check each contact data in DfoActiveContacts to see if we find a match
-	for _, contact := range dmwKnownContacts.Contacts {
-		for _, data := range dfoActiveContacts.Data {
-			dataId, _ := strconv.ParseInt(data.Id, 10, 64)
-			if contact.ContactID == dataId {
-				// TODO: Compare the states - if contact found but states not matched, then we need to update.
-				found = true
-				break
-			}
-		}
+// GetTenantCredentials requests user inputs the tenant's clientId into the console for authentication
+func GetTenantData(inputType string) string {
+	fmt.Printf("Input %v': ", inputType)
+	reader := bufio.NewReader(os.Stdin)
 
-		// If no match is found in DfoActiveContacts, add contact data to deltaList
-		if !found {
-			fmt.Printf("Adding ContactID: [%v] to deltaList\n", contact.ContactID)
+	response, err := reader.ReadString('\n')
+	if err != nil {
+		log.Fatal(err)
+	}
 
-			delta := DmwKnownContact{
-				ContactID: contact.ContactID,
-				MasterContactID: contact.MasterContactID,
-				TenantID: contact.TenantID,
-				StartDate: contact.StartDate,
-				FromAddr: contact.FromAddr,
-				CurrentContactState: contact.CurrentContactState,
-				CurrentContactDate: contact.CurrentContactDate,
-				Direction: contact.Direction,
-				ChannelID: contact.ChannelID,
-				StateIndex: contact.StateIndex,
-				CaseIDString: contact.CaseIDString,
-				DigitalContactState: contact.DigitalContactState,
-				EventID: contact.EventID,
-			}
+	return strings.TrimSuffix(response, "\n")
+}
 
-			deltaList = append(deltaList, delta)
+// Validate verifies that the data input by the user was in a proper type or format
+func Validate(inputValue string, isGuid bool, inputValueValid bool) bool {
+	if isGuid {
+		if _, err := uuid.Parse(inputValue); err != nil {
+			fmt.Printf("Input value is not a valid GUID: [%v]\n", inputValue)
+			return inputValueValid
```

Comment:
Curious why this returns a parameter instead of just 'false' if it's not valid. Same with the other returns in this function.

Reply:
Because I didn't think to just do that! lol  Fixed.